doc/src/manual/metaprogramming.md

```diff
@@ -644,6 +644,61 @@ This allows macros to look up contextual information, such as existing bindings,
 or to insert the value as an extra argument to a runtime function call doing self-reflection
 in the current module.
 
+### Renumbering lines in macros
+
+You will likely want to renumber new lines of code introduced in macros
+Consider the following trivial macro:
+
+```jldoctest macro_line_numbers
+julia> macro adds_more_lines_1()
+           quote
+               "code"
+           end
+       end
+@adds_more_lines_1 (macro with 1 method)
+```
+
+Let's take a closer look at the new code introduced by the macro:
+
+```jldoctest macro_line_numbers
+julia> numbered = quote
+           "code"
+       end;
+
+julia> numbered.head
+:block
+
+julia> typeof.(numbered.args)
+2-element Vector{DataType}:
+ LineNumberNode
+ String
+```
+
+Julia will introduce a new line number for the new line of code pointing to the definition expression.
+You likely want line numbering to point to the user's call site, rather than the definition expression.
+This is for the following two reasons:
+
+- If an error occurs in code introduced in a macro, you likely want stack-traces to point to the user's call site, rather than the definition expression.
```

Comment:
```suggestion
- If an error occurs in code introduced in a macro, you likely want stack-traces to point to the user's call site, rather than the macro's internals.
```